File: scala/485_Max_Consecutive_Ones.scala
object Solution {
    def findMaxConsecutiveOnes(nums: Array[Int]): Int = {
        var maxCount = 0;
        var tempCount = 0;
        for (num <- nums) {
            if (num == 0) {
                maxCount = maxCount.max(tempCount);
                tempCount = 0;
            } else {
                tempCount += 1;
            }
        }
        maxCount = maxCount.max(tempCount);

        return maxCount;
    }
}

File: java/485_Max_Consecutive_Ones.java
class Solution {
    public int findMaxConsecutiveOnes(int[] nums) {
        int maxCount = 0;
        int tempCount = 0;
        
        for (int num : nums) {
            if (num == 0) {
                maxCount = Math.max(maxCount, tempCount);
                tempCount = 0;
            } else {
                tempCount++;
            }
        }
        maxCount = Math.max(maxCount, tempCount);
        
        return maxCount;

    }
}
